crates/sui-framework/sources/governance/validator_cap.move

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module sui::validator_cap {
+    use sui::object::{Self, ID, UID};
+    use sui::transfer;
+    use sui::tx_context::TxContext;
+    friend sui::sui_system;
+    friend sui::validator;
+    friend sui::validator_set;
+
+    #[test_only]
+    friend sui::sui_system_tests;
+    #[test_only]
+    friend sui::rewards_distribution_tests;
+
+    /// The capability object is created when creating a new `Validator` or when the
+    /// validator explicitly creates a new capability object for rotation/revocation.
+    /// The holder address of this object can perform some validator operations on behalf of
+    /// the authorizer validator. Thus, if a validator wants to separate the keys for operation
+    /// (such as reference gas price setting or tallying rule reporting) from fund/staking, it
+    /// could transfer this capability object to another address.
+
+    /// To facilitate rotating/revocation, `Validator` stores the ID of currently valid
+    /// `UnverifiedValidatorOperationCap`. Thus, before converting `UnverifiedValidatorOperationCap`
+    /// to `VerifiedValidatorOperationCap`, verification needs to be done to make sure
+    /// the cap object is still valid.
+    struct UnverifiedValidatorOperationCap has key, store {
+        id: UID,
+        authorizer_validator_address: address,
+    }
+
+    /// Privileged operations require `VerifiedValidatorOperationCap` for permission check.
+    /// This is a Hot Potato, only constructed after successful verification,
+    /// and deconstructed after use.
+    struct VerifiedValidatorOperationCap {
```

Comment:
Naming nit: Weirdly, with the more explicit name, I was finding it difficult to distinguish between these two types in the code (because they only differ by two characters), thoughts on:

- `UnverifiedValidatorOperationCap`
- `ValidatorOperationCap`

Further naming nit -- what does `Operation` stand for in this type?  It seems like it could maybe be `UnverifiedValidatorCap` and `ValidatorCap`, because it's implied that `Cap`abilities guard operations.

Reply:
I will rename to `ValidatorOperationCap`. (in our rust codebase, we do the opposite: `VerifiedXYZ` v.s. `XYZ`).

I prefer not removing `Operation` in the name, because otherwise it would make the `Cap` to powerful. For example, this Cap wouldn't allow validators to claim rewards or move funds. By `operation` I try to mean operational stuff such as tallying rule and gas price setting. However I'm open to any other wording suggestions to better represent the ability

Reply:
> I will rename to `ValidatorOperationCap`. (in our rust codebase, we do the opposite: `VerifiedXYZ` v.s. `XYZ`).

This would be fine too as well, by the way, `ValidatorAdminCap` and `VerifiedValidatorAdminCap`.